plugins/inputs/kube_inventory/kube_state.go

```diff
@@ -122,6 +129,13 @@ var availableCollectors = map[string]func(ctx context.Context, acc telegraf.Accu
 }
 
 func (ki *KubernetesInventory) initClient() (*client, error) {
+	// Even neither are provided, use the default service account.
+	if ki.BearerToken == "" && ki.BearerTokenString == "" {
+		if _, err := os.Stat(defaultServiceAccountPath); err == nil {
+			ki.BearerToken = defaultServiceAccountPath
```

Comment:
It would be best to just read the token file once, and base any decisions on if it is successful, instead of doing a stat and then reading later.

As we now allow for a `func (ki *KubernetesInventory) Init() error {` function, this would be a good place to put the checking and reading.